app/src/test/java/org/mozilla/fenix/components/metrics/MetricControllerTest.kt

```diff
@@ -167,4 +172,41 @@ class MetricControllerTest {
         controller.track(Event.TabMediaPause)
         verify { marketingService1.track(Event.TabMediaPause) }
     }
+
+    @Test
+    fun `topsites fact should convert to the right events`() {
+        var enabled = true
+        val controller = ReleaseMetricController(
+            services = listOf(dataService1),
+            isDataTelemetryEnabled = { enabled },
+            isMarketingDataTelemetryEnabled = { enabled }
+        )
+
+        var fact = Fact(
+            Component.FEATURE_TOP_SITES,
+            Action.INTERACTION,
+            TopSitesFacts.Items.COUNT,
+            "1"
+        )
+
+        assertEquals(controller.factToEvent(fact), Event.HaveTopSites)
+
+        fact = Fact(
+            Component.FEATURE_TOP_SITES,
+            Action.INTERACTION,
+            TopSitesFacts.Items.COUNT,
+            "0"
+        )
+
+        assertEquals(controller.factToEvent(fact), Event.HaveNoTopSites)
+
+        fact = Fact(
+            Component.FEATURE_TOP_SITES,
+            Action.INTERACTION,
+            TopSitesFacts.Items.COUNT,
+            "10"
+        )
+
+        assertEquals(controller.factToEvent(fact), Event.HaveTopSites)
```

Comment:
Looks good, maybe we should just add the case where the count is a negative integer (-4) as well

Reply:
Agreed.  I'll add a test case for negative count.  Thanks,